### PR TITLE
Tweak macos screen switch helper

### PR DESCRIPTION
**Pull request by eltariel**
There seems to be a minimum amount of movement needed to convince Mac OS to switch the active screen. Moving 25/65536 units is less than one screen pixel at 1440p and below. Doubling that to 50 (5 * 10 units) is more than one screen pixel all the way down to 720p, and seems to be sufficient on both of my macbooks.

I also refactored the macos-specific desktop switch helper to its own function to make it a little clearer what is going on, and cleaned up some unused parameters in the main desktop switch function.

### CODE REVIEW

**Comment by hrvach** [comment]
Amazing, thank you - I managed to forget about lower resolutions :) I'll try testing it a bit today and whipping up a new release. Really appreciate your patches.

**Comment by eltariel** [comment]
No worries, I'm happy that I can be of assistance. Building something like this has been on my to-do list for a while and finding this project has saved me so much effort 🤣

This change worked for me on a couple of resolutions (2x3008x1692 scaled on dual 4k monitors, and 2056x1329+3008x1629 between the internal screen and one 4k monitor) where the previous values didn't. Realistically we could probably do a single relative report of 100 units which would be ~2px at 720p and ~6px at 4k assuming I understand the maths correctly, but I haven't actually tried that yet...

File: src/mouse.c [hunk 1]
```diff
@@ -137,17 +140,24 @@ void switch_screen(
     state->pointer_y = scale_y_coordinate(output->number, 1 - output->number, state);
 }
 
-void switch_desktop(device_t *state, output_t *output, int new_index, int direction) {
+#define MACOS_SWITCH_MOVE_X 10
```
Comment (jalmeroth) [inline src/mouse.c:143]: Those two lines seem to be redundant with line 20/21

Comment (eltariel) [reply src/mouse.c]: Whoops, good catch

**Pull request merged**